### PR TITLE
Update scipp neutron docs

### PR DESCRIPTION
**Pull request by OwenArnold**
Ref #1478

Updates to and fixes surrounding `scipp-neutron` in docs

### CODE REVIEW

File: docs/tutorials/neutron-diffraction.ipynb [hunk 1]
```diff
@@ -34,6 +34,8 @@
     "[PG3_4866_event.nxs](http://198.74.56.37/ftp/external-data/MD5/3d543bc6a646e622b3f4542bc3435e7e).\n",
     "Both are available as part of Mantid's test data.\n",
     "\n",
+    "Rename the files upon download.\n",
```
Comment (OwenArnold) [inline docs/tutorials/neutron-diffraction.ipynb:37]: Maybe better to refer to `data/fetch_neutron_data.py` and change back to PG3_4844

File: docs/scipp-neutron/instrument-view.ipynb [hunk 2]
```diff
@@ -62,7 +62,8 @@
    },
    "outputs": [],
    "source": [
-    "sample = sn.load(filename='PG3_4844_event.nxs')"
+    "sample = sn.load(filename='PG3_4871_event.nxs')\n",
```
Comment (OwenArnold) [inline docs/scipp-neutron/instrument-view.ipynb:65]: Maybe better to refer to `data/fetch_neutron_data.py` and change back to PG3_4844

File: docs/scipp-neutron/groupby.ipynb [hunk 3]
```diff
@@ -12,7 +12,11 @@
     "## Grouping with bins\n",
     "\n",
     "Note that this notebooks requires [Mantid](https://www.mantidproject.org/Main_Page) and data files.\n",
-    "A [binder](https://mybinder.org/v2/gh/scipp/scipp-neutron-jupyter-demo/master) is available that can run this notebook."
+    "A [binder](https://mybinder.org/v2/gh/scipp/scipp-neutron-jupyter-demo/master) is available that can run this notebook.\n",
+    "\n",
+    "[PG3_4844_event.nxs](http://198.74.56.37/ftp/external-data/MD5/d5ae38871d0a09a28ae01f85d969de1e) is required for this notebook.\n",
```
Comment (OwenArnold) [inline docs/scipp-neutron/groupby.ipynb:17]: Maybe better to refer to `data/fetch_neutron_data.py`

File: docs/scipp-neutron/from-mantid-to-scipp.ipynb [hunk 4]
```diff
@@ -325,8 +329,8 @@
    "outputs": [],
    "source": [
     "mantid.ConjoinXRuns(\n",
-    "    InputWorkspaces='data1,data2',\n",
-    "    OutputWorkspace='data')"
+    "    InputWorkspaces=['input','input'],\n",
+    "    OutputWorkspace='data')\n"
```
Comment (nvaytet) [inline docs/scipp-neutron/from-mantid-to-scipp.ipynb:333]:
```suggestion
    "    OutputWorkspace='data')"
```

File: docs/scipp-neutron/from-mantid-to-scipp.ipynb [hunk 5]
```diff
@@ -416,17 +431,20 @@
...
    ]
   },
   {
    "cell_type": "markdown",
    "metadata": {},
    "source": [
-    "This requires aligned dimensions (matching coordinates) in all input arrays.\n",
-    "Use a Python `dict` or `list` for grouping unaligned data."
+    "This requires aligned dimensions (matching coordinates) in all input arrays. \n",
+    "It is a more powerful concept that that provided by WorkspaceGroups. \n",
```
Comment (nvaytet) [inline docs/scipp-neutron/from-mantid-to-scipp.ipynb:444]:
```suggestion
    "It is a more powerful concept than that provided by WorkspaceGroups. \n",
```

File: docs/scipp-neutron/from-mantid-to-scipp.ipynb [hunk 6]
```diff
@@ -594,6 +618,21 @@
...
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "<div class=\"alert alert-info\">\n",
+    "\n",
+    "**Note**\n",
+    "\n",
+    "That both scipp and Mantid support binary and inplace operations such as + and +=.\n",
+    "Mantid's binary operations call underlying algorithms as part of implementation.\n",
+    "This makes it difficult to change some default behaviour, for example if you want to prevent output workspaces from being registered in Mantids Analysis Data Service.\n",
```
Comment (nvaytet) [inline docs/scipp-neutron/from-mantid-to-scipp.ipynb:631]:
```suggestion
    "This makes it difficult to change some default behaviour, for example if you want to prevent output workspaces from being registered in Mantid's Analysis Data Service.\n",
```

File: docs/scipp-neutron/from-mantid-to-scipp.ipynb [hunk 7]
```diff
@@ -594,6 +618,21 @@
...
   },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "<div class=\"alert alert-info\">\n",
+    "\n",
+    "**Note**\n",
+    "\n",
+    "That both scipp and Mantid support binary and inplace operations such as + and +=.\n",
```
Comment (nvaytet) [inline docs/scipp-neutron/from-mantid-to-scipp.ipynb:629]:
```suggestion
    "Both scipp and Mantid support binary and in-place operations such as + and +=.\n",
```

File: docs/scipp-neutron/from-mantid-to-scipp.ipynb [hunk 8]
```diff
@@ -594,6 +618,21 @@
...
+    "Mantid's binary operations call underlying algorithms as part of implementation.\n",
```
Comment (nvaytet) [inline docs/scipp-neutron/from-mantid-to-scipp.ipynb:630]:
```suggestion
    "Mantid's binary operations call underlying algorithms as part of their implementation.\n",
```

File: docs/scipp-neutron/from-mantid-to-scipp.ipynb [hunk 9]
```diff
@@ -679,7 +720,19 @@
...
+    "except RuntimeError as err:\n",
+    "    print(str(err))\n",
+    "da += 7.5 * sc.units.us # This is fine now RHS has units"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Mantid does not have this safety"
```
Comment (nvaytet) [inline docs/scipp-neutron/from-mantid-to-scipp.ipynb:735]:
```suggestion
    "Mantid does not have this safety net."
```

File: docs/scipp-neutron/from-mantid-to-scipp.ipynb [hunk 10]
```diff
@@ -784,7 +837,7 @@
    "metadata": {},
    "outputs": [],
    "source": [
-    "dspacing = sc.neutron.convert(tof, 'tof', 'd-spacing')"
+    "sc.neutron.convert(da, 'tof', 'd-spacing')"
    ]
   },
   {
```
Comment (nvaytet) [inline docs/scipp-neutron/from-mantid-to-scipp.ipynb:843]: It seems you modified this notebook so that it would be run when the docs are built.
If so, you need to remove the
```
"nbsphinx": {
   "execute": "never"
  }
```
in the notebook metadata, at the end of the notebook.

File: python/src/scipp/neutron/instrument_view.py [hunk 11]
```diff
@@ -18,7 +18,7 @@ def instrument_view(scipp_obj=None,
     of bin edges) as values.
     If no bins are specified, the data is histogrammed according to the bins
     that wrap the event data when it is loaded
-    (see https://scipp.github.io/user-guide/binned-data.html).
+    (see https://scipp.github.io/user-guide/binned-data.html ).
```
Comment (nvaytet) [inline python/src/scipp/neutron/instrument_view.py:21]: Why this change?

File: docs/scipp-neutron/groupby.ipynb (hunk 3 above)
Comment (nvaytet) [reply docs/scipp-neutron/groupby.ipynb:17]: Yes I don't know what is better... I think there are some hard-coded paths in the script, so not sure it's good to point to it now. Maybe if we make it more generic/flexible, we could change the docs then?

**Review by nvaytet — CHANGES_REQUESTED** [review]

File: docs/scipp-neutron/from-mantid-to-scipp.ipynb (hunk 10 above)
Comment (OwenArnold) [reply docs/scipp-neutron/from-mantid-to-scipp.ipynb:843]: Yes indeed. Thanks for reminding me.

Comment (OwenArnold) [reply docs/scipp-neutron/from-mantid-to-scipp.ipynb:843]: It does depend on having `mantid-framework` available too.

File: python/src/scipp/neutron/instrument_view.py (hunk 11 above)
Comment (OwenArnold) [reply python/src/scipp/neutron/instrument_view.py:21]: Link doesn't work via help as is. Interprets `).` as part of url

File: docs/scipp-neutron/from-mantid-to-scipp.ipynb (hunk 10 above)
Comment (nvaytet) [reply docs/scipp-neutron/from-mantid-to-scipp.ipynb:843]: Yes, but that's fine for the docs

**Comment by nvaytet** [comment]
Still 2 suggestions unresolved

File: docs/scipp-neutron/from-mantid-to-scipp.ipynb [hunk 12]
```diff
@@ -308,7 +312,7 @@
    "metadata": {},
    "outputs": [],
    "source": [
-    "data = sc.concatenate(data1, data2, 'spectrum')"
+    "data = sc.concatenate(da, da, 'spectrum')"
```
Comment (SimonHeybrock) [inline docs/scipp-neutron/from-mantid-to-scipp.ipynb:315]: Using same input twice may be confusing?

File: docs/scipp-neutron/from-mantid-to-scipp.ipynb [hunk 13]
```diff
@@ -325,7 +329,7 @@
    "outputs": [],
    "source": [
     "mantid.ConjoinXRuns(\n",
-    "    InputWorkspaces='data1,data2',\n",
+    "    InputWorkspaces=['input_point_data','input_point_data'],\n",
```
Comment (SimonHeybrock) [inline docs/scipp-neutron/from-mantid-to-scipp.ipynb:332]: Don't use same input twice?

File: docs/scipp-neutron/from-mantid-to-scipp.ipynb [hunk 14]
```diff
@@ -342,7 +346,7 @@
    "metadata": {},
    "outputs": [],
    "source": [
-    "data = sc.concatenate(data1, data2, 'tof')"
+    "sc.concatenate(da, da, 'tof')"
```
Comment (SimonHeybrock) [inline docs/scipp-neutron/from-mantid-to-scipp.ipynb:349]: Same input twice?

File: docs/scipp-neutron/from-mantid-to-scipp.ipynb [hunk 15]
```diff
@@ -359,11 +363,9 @@
    "outputs": [],
    "source": [
     "mantid.ConjoinSpectra(\n",
-    "    InputWorkspaces='ws1,ws2',\n",
-    "    OutWorkspace='spec_vs_temperature'\n",
-    "    WorkspaceIndex=7,\n",
-    "    LabelUsing=\"temperature\",\n",
-    "    LabelValue=\"Mean\")"
+    "    InputWorkspaces='input_point_data, input_point_data',\n",
```
Comment (SimonHeybrock) [inline docs/scipp-neutron/from-mantid-to-scipp.ipynb:366]: Same input twice?

File: docs/scipp-neutron/from-mantid-to-scipp.ipynb [hunk 16]
```diff
@@ -379,10 +381,23 @@
...
+    "Or more efficiently"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "sc.concatenate(da['spectrum', 7], da['spectrum', 7], 'spectrum')"
```
Comment (SimonHeybrock) [inline docs/scipp-neutron/from-mantid-to-scipp.ipynb:400]: Same input twice

File: docs/scipp-neutron/from-mantid-to-scipp.ipynb [hunk 17]
```diff
@@ -399,7 +414,7 @@
    "outputs": [],
    "source": [
     "mantid.GroupWorkspaces(\n",
-    "    InputWorkspaces='ws1,ws2',\n",
+    "    InputWorkspaces='input_point_data,input_point_data',\n",
```
Comment (SimonHeybrock) [inline docs/scipp-neutron/from-mantid-to-scipp.ipynb:417]: Same input twice

File: docs/scipp-neutron/from-mantid-to-scipp.ipynb [hunk 18]
```diff
@@ -416,17 +431,20 @@
    "metadata": {},
    "outputs": [],
    "source": [
-    "data = sc.Dataset({\n",
-    "    'data1':data_array1,\n",
-    "    'data2':data_array2})"
+    "sc.Dataset({\n",
+    "    'data1':da,\n",
+    "    'data2':da})"
```
Comment (SimonHeybrock) [inline docs/scipp-neutron/from-mantid-to-scipp.ipynb:436]: Same input twice.

File: docs/scipp-neutron/from-mantid-to-scipp.ipynb [hunk 19]
```diff
@@ -416,17 +431,20 @@
...
    ]
   },
   {
    "cell_type": "markdown",
    "metadata": {},
    "source": [
-    "This requires aligned dimensions (matching coordinates) in all input arrays.\n",
-    "Use a Python `dict` or `list` for grouping unaligned data."
+    "This requires aligned dimensions (matching coordinates) in all input arrays. \n",
+    "It is a more powerful concept than that provided by WorkspaceGroups. \n",
```
Comment (SimonHeybrock) [inline docs/scipp-neutron/from-mantid-to-scipp.ipynb:444]: Maybe we should say "more powerful but restricted" (or limited?)?

File: docs/scipp-neutron/from-mantid-to-scipp.ipynb [hunk 20]
```diff
@@ -465,7 +483,8 @@
     "    ['tof'],\n",
     "    values=np.arange(0.0, 20000.0, 100.0),\n",
     "    unit=sc.units.us)\n",
-    "histo = sc.rebin(histo, 'tof', edges)"
+    "data = sc.compat.mantid.from_mantid(input_bin_edges)\n",
```
Comment (SimonHeybrock) [inline docs/scipp-neutron/from-mantid-to-scipp.ipynb:486]: `sc.neutron.from_mantid`?

File: docs/scipp-neutron/from-mantid-to-scipp.ipynb [hunk 21]
```diff
@@ -114,7 +117,9 @@
    "metadata": {},
    "outputs": [],
    "source": [
-    "copy = data.copy()"
+    "da = sc.compat.mantid.from_mantid(input_point_data)\n",
```
Comment (SimonHeybrock) [inline docs/scipp-neutron/from-mantid-to-scipp.ipynb:120]: `sc.neutron.from_mantid`?

File: docs/scipp-neutron/from-mantid-to-scipp.ipynb (hunk 12 above)
Comment (nvaytet) [reply docs/scipp-neutron/from-mantid-to-scipp.ipynb:315]: I agree that 2 different inputs would be better. I think it's standard for developers who understand that you often do this when you are testing stuff (e.g. in unit tests) but users might be confused and think as to what is trying to be achieved here, and some may wonder if it only works with twice the same input ?

Comment (OwenArnold) [reply docs/scipp-neutron/from-mantid-to-scipp.ipynb:315]: Completely agree with the statement above. I'll change to something meaningful.

File: docs/scipp-neutron/groupby.ipynb (hunk 3 above)
Comment (OwenArnold) [reply docs/scipp-neutron/groupby.ipynb:17]: I'm going to leave this as is

File: docs/scipp-neutron/from-mantid-to-scipp.ipynb [hunk 22]
```diff
@@ -784,16 +855,42 @@
...
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "<div class=\"alert alert-info\">\n",
+    "\n",
+    "**Note**\n",
+    "\n",
+    "The `scipp.neutron module` also provides a `to_mantid` function, which has limited support for converting scipp data to Mantid workspaces. Because scipp offers a more flexible container than a Workspace, in particular MatrixWorkspace, it is not always possible to exactly convert all information to Mantid workspaces.\n",
```
Comment (nvaytet) [inline docs/scipp-neutron/from-mantid-to-scipp.ipynb:883]: Can you split long lines at the end of each sentence?
The convention we adopted is one sentence per line.

Comment (OwenArnold) [reply docs/scipp-neutron/from-mantid-to-scipp.ipynb:883]: Indeed, will fix

**Review by nvaytet — APPROVED** [review]

**Comment by SimonHeybrock** [comment]
There is a conflict in the instrument view now.

**Comment by SimonHeybrock** [comment]
@OwenArnold Builds are failing.

File: python/src/scipp/neutron/instrument_view.py [hunk 23]
```diff
@@ -9,13 +9,19 @@ def instrument_view(scipp_obj=None, positions="position", **kwargs):
     detector vector positions.
     Use the `positions` argument to change the vectors used as pixel positions.
     Sliders are added to navigate extra dimensions.
-    Spatial slicing and pixel opacity control is available using the controls
-    below the scene.
-    Use the `pixel_size` argument to specify the size of the detectors.
-    If no `pixel_size` is given, a guess is performed based on the extents of
-    the positions.
-    The aspect ratio of the positions is preserved by default, but this can
-    be changed to automatic scaling using `aspect="equal"`.
```
Comment (nvaytet) [inline python/src/scipp/neutron/instrument_view.py:18]: So i updated the doc string in my PR (https://github.com/scipp/scipp/pull/1537/files#diff-363ad19e0ca6011d877b557e14ece009d135089a0349acea5548fb447d50839a) but this reverts it to the previous state.

**Review by nvaytet — CHANGES_REQUESTED** [review]

File: docs/scipp-neutron/from-mantid-to-scipp.ipynb (hunk 20 above)
Comment (nvaytet) [reply docs/scipp-neutron/from-mantid-to-scipp.ipynb:504]: i agree that using `neutron` is better than introducing a new `compat` submodule here

**Review by nvaytet — CHANGES_REQUESTED** [review]

**Review by nvaytet — APPROVED** [review]
